src/ZX/rules.jl

```diff
@@ -912,19 +930,26 @@ function rewrite!(::Rule{:id}, zxg::ZXGraph{T, P}, vs::Vector{T}) where {T, P}
         set_phase!(zxg, v2, zero(P))
         set_phase!(zxg, v1, -phase(zxg, v1))
         zxg.phase_ids[v1] = (zxg.phase_ids[v1][1], -zxg.phase_ids[v1][2])
+      end
+      if ((spider_type(zxg, v1) == SpiderType.In || spider_type(zxg, v1) == SpiderType.Out) && 
+          (spider_type(zxg, v3) == SpiderType.In || spider_type(zxg, v3) == SpiderType.Out))
+          rem_spider!(zxg, v2)
+          add_edge!(zxg, v1, v3, EdgeType.SIM)
```

Comment:
Do you need to check the edge type for the edge (v1, v2) and (v2, v3)? For example, if only one of them are Hadamard edge, then the resulting edge should also be a Hadamard edge.

Reply:
Your assumption seems to be correct. I replaced the and with and or